data/core.yaml

```diff
@@ -735,6 +735,7 @@ en:
     lock:
       suggestion: 'The "{label}" field is locked because there is a Wikidata tag. You can delete it or edit the tags in the "Tags" section.'
     display_name:
+      ref_name: "{ref}: {name}"
```

Comment:
Should there be a `network_ref_name` string in case the `network` is also tagged?

/ref https://github.com/openstreetmap/iD/issues/8559#issuecomment-871020386

Reply:
I wasn't sure if your comment still applied, since the network name is sometimes shown twice in the latest version of iD.

If this is intentional then I'll make the change to also have `network_ref_name`

![image](https://user-images.githubusercontent.com/16009897/134105659-3a10aa0d-b2b1-471c-a71f-5f705b9adebd.png)

Reply:
Ah, that’s because Auckland Transport has an entry in the name suggestion index. Whenever NSI has an entry, it would make sense to hide the redundant, unpolished `network` value. However, omitting `network` would lead a regression of #3301 for any network that isn’t included in NSI.

This function is called from a couple different places. The relation membership editor happens to prepend the preset name. But `utilDisplayNameForPath()` calls this function without prepending the preset name, and some other places only use the preset name as a fallback if there’s no display name:

https://github.com/openstreetmap/iD/blob/fa5e7bca20ed38da3ae0ef87f7d3d7e7bdfd4827/modules/ui/keepRight_details.js#L105-L110

This function could take an additional parameter indicating whether to display the network if available. In the relation membership editor, that flag could be `false` if `matched.name()` is defined:

https://github.com/openstreetmap/iD/blob/fa5e7bca20ed38da3ae0ef87f7d3d7e7bdfd4827/modules/ui/sections/raw_membership_editor.js#L345-L346

Reply:
> In the relation membership editor, that flag could be `false` if `matched.name()` is defined:

That won't work because `matched.name()` seems to always be defined, it's either a name from NSI like `AT (Auckland Transport)` or a generic name like `Bus Route`. I used `matched.reference().qid` instead, is that reliable?

Here's the result now: the bottom one has `network:wikidata`, the top one doesn't:
![image](https://user-images.githubusercontent.com/16009897/134291587-0e33b17b-7b84-4386-a7a9-f98325d1456c.png)

Reply:
`reference()` is too broad, potentially returning a QID if the route happens to be tagged with `wikidata` or `operator:wikidata`, whereas directly referencing `matched.tags['network:wikidata']` would be more accurate:

https://github.com/openstreetmap/iD/blob/fa5e7bca20ed38da3ae0ef87f7d3d7e7bdfd4827/modules/presets/preset.js#L156-L164

Unfortunately, NSI has quite a few transit route entries that lack `network:wikidata` tags ([example](https://nsi.guide/index.html?t=transit&k=route&v=bus#aleopenvendee-ed7458)).

How about checking for `matched.tags.network` instead? After all, we’re trying to avoid duplicating `network`. Every transit route entry in NSI should have a `network` tag. (You can verify that’s the case using `jq '.items | map(select(.tags | has("network") | not))' data/transit/route/*.json` in a clone of NSI.) @bhousel, would this be a safe assumption?

Reply:
> Unfortunately, NSI has quite a few transit route entries that lack `network:wikidata` tags ([example](https://nsi.guide/index.html?t=transit&k=route&v=bus#aleopenvendee-ed7458)).

I'm pretty sure that the ones without a wikidata tag won't get presets generated for them and then won't ever match a preset in this code.

You can also check whether `matched.suggestion = true`, that will be set if it's a name-suggestion-index preset.

Reply:
Thanks, `matched.suggestion` works well, I've changed it to that